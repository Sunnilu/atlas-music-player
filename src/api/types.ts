// src/api/types.ts
export interface Song {
  url: any;
  audio: any;
  image: string;
  author: string;
  id: number;
  title: string;
  artist: string;
  genre: string;
  duration: string;
  cover: string;
  song: string;
}

export interface ApiResponse<T> {
  status: 'success' | 'error';
  data?: T;
  error?: string;
}