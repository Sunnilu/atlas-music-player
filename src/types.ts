// src/types.ts
export interface Song {
  id: number;
  title: string;
  author: string;      // 🔁 renamed from `artist`
  duration: string;
  image: string;
  lyrics?: string;
  audio: string;
}
export {};
