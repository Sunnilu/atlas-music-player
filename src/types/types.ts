// src/types.ts
export interface Song {
    id: number;
    title: string;
    artist: string;
    duration: string;
    image: string;   // URL for album artwork
    lyrics?: string; // Optional lyrics field
  }
  