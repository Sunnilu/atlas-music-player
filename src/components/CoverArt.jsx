// src/components/CoverArt.jsx
import { useState } from 'react';
import { Music } from 'react';

export default function CoverArt() {
  return (
    <div className="w-48 h-48 mx-auto mb-6 bg-gray-100 rounded-lg flex items-center justify-center">
      <Music className="w-12 h-12 text-gray-400" />
    </div>
  );
}
