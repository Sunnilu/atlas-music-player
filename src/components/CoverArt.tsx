// src/components/CoverArt.tsx
import React, { useState } from 'react';
import { Song } from '@types';

interface CoverArtProps {
  song: Song | null;
}

const CoverArt: React.FC<CoverArtProps> = ({ song }) => {
  const [showLyrics, setShowLyrics] = useState(false);

  if (!song) return null;

  return (
    <div
      className="relative w-full aspect-square rounded-xl overflow-hidden shadow-md"
      onMouseEnter={() => setShowLyrics(true)}
      onMouseLeave={() => setShowLyrics(false)}
    >
      <img
        src={song.image}
        alt={`${song.title} cover`}
        className="w-full h-full object-cover"
      />

      {showLyrics && song.lyrics && (
        <div className="absolute inset-0 bg-black bg-opacity-80 text-white text-sm p-4 overflow-y-auto">
          <p className="whitespace-pre-wrap">{song.lyrics}</p>
        </div>
      )}
    </div>
  );
};

export default CoverArt;
