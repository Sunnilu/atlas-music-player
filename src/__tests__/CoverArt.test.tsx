import { describe, it, expect } from 'vitest';
import renderer from 'react-test-renderer';
import CoverArt from '../components/CoverArt';
import { Song } from '../types';

const mockSong: Song = {
  id: 1,
  title: 'Test Track',
  artist: 'Atlas',
  audio: '/test.mp3',
  image: '/cover.jpg', // âœ… Corrected
  duration: '3:00',
  lyrics: '',
  cover: undefined,
  author: ''
};

describe('CoverArt Snapshot', () => {
  it('renders with image and no lyrics', () => {
    const tree = renderer.create(<CoverArt song={mockSong} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

