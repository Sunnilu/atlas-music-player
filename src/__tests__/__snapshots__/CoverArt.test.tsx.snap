// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt Snapshot > renders with image and no lyrics 1`] = `null`;
